sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -276,11 +276,34 @@ def _create(self, resource_type, content_settings=None, metadata=None, **kwargs)
             The timeout parameter is expressed in seconds.
         :return: Dict[str, Union[str, datetime]]
         """
-        options = self._create_path_options(
-            resource_type,
-            content_settings=content_settings,
-            metadata=metadata,
-            **kwargs)
+        lease_id = kwargs.get('lease_id', None)
+        lease_duration = kwargs.get('lease_duration', None)
+        expires_on = kwargs.get('expires_on', None)
+        if lease_id and not lease_duration:
+            raise ValueError("Please specify a lease_id and a lease_duration.")
+        if lease_duration and not lease_id:
+            raise ValueError("Please specify a lease_id and a lease_duration.")
+        if expires_on:
```

Comment:
Not sure if this is the best way to approach this. Open to suggestions that would be logically equivalent! Just need to somehow conditionally set `expiry_options` (if not `None`) and pass this onto this `_create_path_options` API call

Reply:
I think you should do all the work in `_create_path_options` itself. Let `expires_on` go into `_create_path_options` normally and do the type check in there and set the `expiry_options` appropriately. (This gets the added benefit of being shared with the `async` client).

Also don't forget to update the docstring for this method as well :).

Reply:
Ah okay that makes sense!

Good catch! 😄